data-avro/src/test/java/com/linkedin/data/avro/TestDataTranslator.java

```diff
@@ -1710,6 +1710,32 @@ private void testDataTranslation(String schemaText, String[][] row) throws IOExc
             false,
             "",
         },
+        // If complex field in pegasus schema has default value and it doesn't specify value for a nested required field,
+        // then the default value specified at field level should be used.
+        {
+            // required int with default value
+            "{ \"type\" : \"record\", \"name\" : \"foo\", \"fields\" : [ "
+                + "{ \"name\" : \"bar\", \"type\" : "
+                + "{ \"type\": \"record\", \"name\": \"Bar\", \"fields\": ["
+                + " { \"name\": \"barInt\", \"type\": ##T_START \"int\" ##T_END, \"default\" : 42 }"
+                + " ] },"
+                + " \"default\": {}"
+                + " } ] }",
+            PegasusToAvroDefaultFieldTranslationMode.TRANSLATE,
+            "{ \"type\" : \"record\", \"name\" : \"foo\", \"fields\" : [ "
+                + "{ \"name\" : \"bar\", \"type\" : "
+                + "{ \"type\": \"record\", \"name\": \"Bar\", \"fields\": ["
+                + " { \"name\": \"barInt\", \"type\": \"int\", \"default\" : 42 }"
+                + " ] },"
+                + " \"default\": { \"barInt\":42 }"
+                + " } ] }",
+            "{\"bar\":{\"barInt\":42}}",
+            PegasusToAvroDefaultFieldTranslationMode.TRANSLATE,
+            "{\"bar\":{\"barInt\":42}}",
+            false,
+            "",
+        },
+
 
         // Below are test case example for invalid option combination:
         // If the field in pegasus schema has been translated as required, its data cannot be translated as optional
```

Comment:
Shall we also add a negative test case which will fall into the exception path: throw new IllegalArgumentException(
            message(path, "Cannot translate required field without default."))?

Reply:
It is not possible to test that case as PDL schema validation will fail for the same reason (I tried it).